napari/_qt/widgets/qt_viewer_buttons.py

```diff
@@ -204,16 +204,19 @@ def _open_roll_popup(self):
         if self.viewer.dims.ndisplay != 2:
             return
 
-        dim_sorter = QtDimsSorter(self.viewer, self)
+        # popup
+        pop = QtPopup(self)
```

Comment:
I will feel more comfortable if we will store references to `pop` in class instances. Otherwise GC may clean it to early.

Reply:
Do you mean store it as `QtViewerButtons._roll_popup`? Then discard it on the `QDialog.finished` signal?

Note that other viewer button popups here (grid, perspective) use the same pattern as this. So maybe this can be done in a follow-up PR for all of them?